src/klein/app.py

```diff
@@ -198,16 +200,33 @@ def index(request):
 
         @returns: decorated handler function.
         """
-        segment_count = self._segments_in_url(url) + self._subroute_segments
 
-        @named("router for '" + url + "'")
+        def _urlToString(_url):
+            return _url.asText()
+
+        if isinstance(url, string_types + (binary_type, )):
+            url = URL.fromText(url)
```

Comment:
The percent-encoding described here is implemented by URL.  In fact it is the entire point of URL, to enable Twisted to handle such URLs correctly :).

If you look at the bytes your browser is actually sending on the wire, you will see that the actual text sent suggests that the URL is in fact `https://facebook.com/pages/%E5%A4%A7%E9%98%AA%E5%BA%9C%E5%A4%A7%E6%9D%B1%E5%B8%82%E7%AB%8B%E5%9B%9B%E6%9D%A1%E4%B8%AD%E5%AD%A6%E6%A0%A1/434035363318098` and not the text that you pasted.  If you paste that giant mess of percent-encoding into a browser, you will see it transformed magically into the non-ASCII characters above.